actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -165,15 +165,28 @@ module CacheHelper
       # expire the cache.
       def cache(name = {}, options = {}, &block)
         if controller.respond_to?(:perform_caching) && controller.perform_caching
-          name_options = options.slice(:skip_digest)
-          safe_concat(fragment_for(cache_fragment_name(name, **name_options), options, &block))
+          CachingRegistry.track_caching do
+            name_options = options.slice(:skip_digest)
+            safe_concat(fragment_for(cache_fragment_name(name, **name_options), options, &block))
+          end
         else
           yield
         end
 
         nil
       end
 
+      # Returns whether the current view fragment is within a +cache+ block.
```

Comment:
If we want `uncachable!` for this too, I'd define an exception atop this helper:

```ruby
module CacheHelper
  class UncachableFragmentError < StandardError; end
  …
```

And then define the method above `caching?` like this:

```ruby
# Raises +UncachableFragmentError+ when called from within a +cache+ block.
#
# Useful to denote helper methods that can't participate in fragment caching:
#
#   def project_name_with_time(project)
#     uncachable!
#     "#{project.name} - #{Time.now}"
#   end
#
#   # Which will then raise if used within a +cache+ block:
#   <% cache project do %>
#     <%= project_name_with_time(project) %>
#   <% end %>
def uncachable!
  raise UncachableFragmentError, "can't be fragment cached" if caching?
end
```